nixos/modules/services/misc/n8n.nix

```
@@ -19,47 +26,70 @@ in
       description = "Open ports in the firewall for the n8n web interface.";
     };
 
-    settings = lib.mkOption {
-      type = format.type;
-      default = { };
-      description = ''
-        Configuration for n8n, see <https://docs.n8n.io/hosting/environment-variables/configuration-methods/>
-        for supported values.
-      '';
-    };
-
-    webhookUrl = lib.mkOption {
-      type = lib.types.str;
-      default = "";
+    environment = lib.mkOption {
       description = ''
-        WEBHOOK_URL for n8n, in case we're running behind a reverse proxy.
-        This cannot be set through configuration and must reside in an environment variable.
+        Environment variables to pass to the n8n service.
+        See <https://docs.n8n.io/hosting/configuration/environment-variables/> for available options.
       '';
+      type = lib.types.submodule {
+        freeformType = with lib.types; attrsOf str;
+        options = {
+          GENERIC_TIMEZONE = lib.mkOption {
+            type = lib.types.str;
+            default = if (config.time.timeZone != null) then config.time.timeZone else "America/New_York";
+            defaultText = lib.literalExpression ''
+              if config.time.timeZone
```

What about leave this option unset/null instead of hard coding n8n's default?
```suggestion
              if (config.time.timeZone != null)
```

---

I thought allowing null would make environement have `GENERIC_TIMEZONE=null`. But it doesn't so changed it to the simpler `default = config.time.timeZone` 👍🏽